pkg/object/qingstor.go

```diff
@@ -255,23 +255,27 @@ func newQingStor(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Invalid endpoint: %v, error: %v", endpoint, err)
 	}
-	hostParts := strings.SplitN(uri.Host, ".", 3)
-	bucketName := hostParts[0]
-	zone := hostParts[1]
-
+	var bucketName, zone, host string
+	if !strings.HasSuffix(uri.Host, "qingstor.com") {
+		// support private cloud
+		hostParts := strings.SplitN(uri.Host, ".", 2)
+		bucketName, zone, host = hostParts[0], "", hostParts[1]
+	} else {
+		hostParts := strings.SplitN(uri.Host, ".", 3)
+		bucketName, zone, host = hostParts[0], hostParts[1], hostParts[2]
```

Comment:
Should the zone be part of host? Please add an example here.

Reply:
This is old logic and has not changed.

endpoint="http://zhjijian-test.sh1a.qingstor.com"

![image](https://user-images.githubusercontent.com/31313340/149777429-bd0cae9f-3329-494f-9ef2-6094b81ef620.png)

![image](https://user-images.githubusercontent.com/31313340/149776755-01b16c79-5971-4f40-9b4b-96dd5f35f125.png)

The zone is indeed part of the host.

Reply:
I meant the `host` should be `sh1a.qingstor.com`, not `qingstor.com`, right?

Reply:
### host does not include region
![image](https://user-images.githubusercontent.com/31313340/149790234-de2642da-a6a0-4397-be36-273cd56d0f44.png)